### PR TITLE
Allow full deletes of Required Signoffs, and cancelling deletions

### PR DESCRIPTION
**Pull request by mozbhearsum**
This implements one-click deletions for Required Signoffs via the SpeedDial button (you can already do this, but you have to click the trash icon on each signoff on the page).

While testing this, I also noticed we never wired up the "Cancel Delete" button on `ListSignoffs`, so I did that as well.

### CODE REVIEW

File: src/views/RequiredSignoffs/ViewSignoff/index.jsx [hunk 1]
```diff
@@ -165,8 +165,23 @@ function ViewSignoff({ isNewSignoff, ...props }) {
...
+      roles: [],
+      originalRoles,
+      additionalRoles: [],
+      isNewSignoff,
+    });
+
+    if (!error) {
+      props.history.push('/required-signoffs');
+    }
+  };
```
Comment (helfi92) [inline src/views/RequiredSignoffs/ViewSignoff/index.jsx]: Any chance we could have the delete preceded with a confirmation dialog?

Comment (mozbhearsum) [reply src/views/RequiredSignoffs/ViewSignoff/index.jsx]: Yeah, that's probably a good idea given that it's a pretty significant event. Will do.

**Comment by mozbhearsum** [comment]
The latest changes work, but it's caused this error: `Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function`. I traced it down to the fact that https://github.com/mozilla-frontend-infra/balrog-ui/blob/master/src/components/DialogAction/index.jsx#L66 happens after the redirect is initiated, but I haven't been able to come up with a way to fix it. If there's some way to do the redirect in `onClose` that would probably work, but I'm not sure it's possible to distinguish between a user initiated close and a post-complete close?

File: src/views/RequiredSignoffs/ViewSignoff/index.jsx [hunk 2]
```diff
@@ -165,8 +178,37 @@ function ViewSignoff({ isNewSignoff, ...props }) {
     }
   };
 
-  // TODO: Add delete logic
-  const handleSignoffDelete = () => {};
+  const handleSignoffDelete = () =>
+    setDialogState({
+      ...dialogState,
+      open: true,
+      title: 'Delete Required Signoffs?',
```
Comment (helfi92) [inline src/views/RequiredSignoffs/ViewSignoff/index.jsx]: Should `Signoffs` be `Signoff`?

Comment (mozbhearsum) [reply src/views/RequiredSignoffs/ViewSignoff/index.jsx]: Personally, I don't think so. We're deleting the entire collection of signoff requirements (that is to say, all of the roles that are currently required), so I think the plural is appropriate.

**Comment by mozbhearsum** [comment]
I'm not 100% certain this is what you had in mind, but it appears to work, and seems to make sense based on my limited knowledge of the cleanup functions.

File: src/components/DialogAction/index.jsx [hunk 3]
```diff
@@ -44,6 +44,12 @@ function DialogAction(props) {
     destructive,
     ...rest
   } = props;
+  let cancelled = false;
+
+  // Set this so we can avoid calling setActionExecuting if the caller
+  // decides to do something that unmounts us in onComplete.
+  useEffect(() => () => (cancelled = true));
```
Comment (helfi92) [inline src/components/DialogAction/index.jsx]: Could you check if it still works if this is rewritten as `useEffect(() => () => (cancelled = true), []);` to not have the empty useEffect triggered on each render.

File: src/components/DialogAction/index.jsx [hunk 4]
```diff
@@ -63,7 +69,9 @@ function DialogAction(props) {
       onComplete(result);
     }
 
-    setActionExecuting(false);
+    if (!cancelled) {
+      setActionExecuting(false);
+    }
```
Comment (helfi92) [inline src/components/DialogAction/index.jsx]: This is exactly what I had in mind :) We should probably also wrap the other `setActionExecuting` with the if statement.

Comment (mozbhearsum) [reply src/components/DialogAction/index.jsx]: Good point. We don't have any onError handlers that would unmount at the moment, but we certainly could in the future...

File: src/components/DialogAction/index.jsx (hunk 3 above)
Comment (mozbhearsum) [reply src/components/DialogAction/index.jsx]: It looks like this doesn't work. With this form of it, I'm guessing the cleanup handler only exists during the first render, and then disappears.

Comment (helfi92) [reply src/components/DialogAction/index.jsx]: Hm, weird :( https://stackoverflow.com/a/55020668/6150432 suggests that it should run. Does it print a console log statement if you put one in the cleanup function?

Comment (mozbhearsum) [reply src/components/DialogAction/index.jsx]: I'm going to go ahead and merge this as is, but if there's something we should look at as a follow-up to improve this, I'm happy to do so.

**Pull request merged**